### PR TITLE
Release/4.32.0-beta1

### PR DESCRIPTION
**Pull request by MarkVillacampa**
### StoreKit 2 Beta

This beta introduces a new configuration option to enable full StoreKit 2 flow on the SDK and the RevenueCat backend.

We have been testing StoreKit 2 support in parallel to StoreKit 1 in our backend for a while and we believe it is ready for widespread use.

If your app is currently using StoreKit 1, it is safe to update to StoreKit 2 and it's even possible to switch back to StoreKit 1 if needed. Switching to StoreKit 2 will not prevent purchases made with StoreKit 1 from being processed.

In order to enable StoreKit 2, add `.with(storeKitVersion: .storeKit2)` to your RevenueCat configuration code:

```
Purchases.configure(with: .builder(withAPIKey: apiKey)
    .with(storeKitVersion: .storeKit2)
    .build()
```

If you were previously using the deprecated configuration option `.with(usesStoreKit2IfAvailable: true)`, we recommend you remove it and switch to the new supported method.

⚠️ ⚠️ Important ⚠️ ⚠️

In order to validate StoreKit 2 purchases, make sure you have an In-App Purchase Key configured in your app.

Please see https://rev.cat/in-app-purchase-key-configuration for more info.

🚧🚧 Limitations 🚧🚧

- Observer Mode is not currently supported when using StoreKit 2.
- The `originalApplicationVersion` and `originalPurchaseDate` properties in `CustomerInfo` are not supported in this first beta. Please do not update if your implementation relies on them being present.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Those paywall changes are already on the last release. I guess you branched off main before that. Maybe just remove them from the release notes?

File: RevenueCat.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = "RevenueCat"
-  s.version          = "4.32.0-SNAPSHOT"
+  s.version          = "4.32.0-beta1"
```
Comment (NachoSoto) [inline RevenueCat.podspec:3]: The convention is:
```suggestion
  s.version          = "4.32.0-beta.1"
```

File: Tests/InstallationTests/CommonFiles/RevenueCat-Swift.h [hunk 2]
```diff
@@ -40,12 +40,18 @@
 #include <string.h>
 #endif
 #if defined(__cplusplus)
-#if __has_include(<ptrauth.h>)
+#if defined(__arm64e__) && __has_include(<ptrauth.h>)
```
Comment (NachoSoto) [inline Tests/InstallationTests/CommonFiles/RevenueCat-Swift.h:43]: What version of Xcode did you use? We still make releases with 14.3 in CI, I guess this was made locally with 15.x, but it should be fine.

**Comment by MarkVillacampa** [comment]
Closed and reopened https://github.com/RevenueCat/purchases-ios/pull/3536 with correct beta naming convention

File: CHANGELOG.latest.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
-### RevenueCatUI
-* `Paywalls`: add `PaywallFooterViewController` (#3486) via Toni Rico (@tonidero)
-* `Paywalls`: improve landscape support of all templates (#3471) via NachoSoto (@NachoSoto)
-* `Paywalls`: ensure footer links open in full-screen sheets (#3524) via NachoSoto (@NachoSoto)
-* `Paywalls`: improve `FooterView` text alignment (#3525) via NachoSoto (@NachoSoto)
-* Paywalls: Add dismissal method in `PaywallViewControllerDelegate` (#3493) via Toni Rico (@tonidero)
+### StoreKit 2 Beta
+
+This beta introduces a new configuration option to enable full StoreKit 2 flow on the SDK and the RevenueCat backend.
```
Comment (aboedo) [inline CHANGELOG.latest.md:3]: let's add some extra clarification between what the new version of SK2 support is vs the old one. We internally call it "full" StoreKit 2, but the distinction probably wont' be immediately clear to a customer, who could reasonably wonder what this means if they've had a storekit 2 flag for years

Comment (aboedo) [reply CHANGELOG.latest.md:3]: That'll also help explain and justify the deprecation of the old flag

File: CHANGELOG.latest.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+⚠️ ⚠️ Important ⚠️ ⚠️
+
+In order to validate StoreKit 2 purchases, make sure you have an In-App Purchase Key configured in your app.
+
+Please see https://rev.cat/in-app-purchase-key-configuration for more info.
+
+🚧🚧 Limitations 🚧🚧
+
+- Observer Mode is not currently supported when using StoreKit 2.
+- The `originalApplicationVersion` and `originalPurchaseDate` properties in `CustomerInfo` are not supported in this first beta. Please do not update if your implementation relies on them being present.
```
Comment (aboedo) [inline CHANGELOG.latest.md:28]: This focuses a lot on the limitations, but there aren't really any mentions of the benefits, right? 
These would include faster purchase processing, less edge cases (cough, invalid receipt, cough) and more accurate intro eligibility calculations, for starters

**Review by aboedo — COMMENTED** [review]
So excited for this! Left some thoughts for the release notes

**Pull request closed without merging**